lib/yard/cli/yardoc.rb

```diff
@@ -294,7 +295,16 @@ def parse_arguments(*args)
         # Last minute modifications
         self.files = Parser::SourceParser::DEFAULT_PATH_GLOB if files.empty?
         files.delete_if {|x| x =~ /\A\s*\Z/ } # remove empty ones
-        readme = Dir.glob('README{,*[^~]}').first
+        readmes = Dir.glob('README{,*[^~]}')
+        if readmes.empty?
+          readme = Dir.glob(files.first).first if options.onefile
```

Comment:
If I see correctly, this branch is doubled by line no. 308.

Reply:
This is correct. Seems like this can be golfed down to just:

```ruby
readme = Dir.glob('README{,*[^~]}').map(....)....first
```

Reply:
I didn't make that change because it goes outside the scope of the original issue (as it deals with a separate optimization / case). If you'd like me to incorporate that change into this patch, I'd be happy to update it.

Reply:
Nevermind, I get it now. You're right.

Reply:
I'll add a test.